### PR TITLE
Implementing new Property

### PR DESCRIPTION
**Pull request by louis1706**
Making than Every 
Pickup 
Item
Projectile 

Share and keep the information of their respective property
for example Item::Scale will keep their change with Pickup.Scale and Projectile.Scale

### CODE REVIEW

File: Exiled.API/Features/Items/Armor.cs [hunk 1]
```diff
@@ -84,32 +85,36 @@ public bool RemoveExcessOnDrop
...
         public int HelmetEfficacy
         {
             get => Base.HelmetEfficacy;
             set
             {
-                if (value is <= 101 and >= 0)
-                    Base.HelmetEfficacy = value;
-                else
-                    throw new ArgumentOutOfRangeException(nameof(HelmetEfficacy), "Value of armor efficacy must be between 0 and 100.");
+                if (value is > 101 or < 0)
```
Comment (Misaka-ZeroTwo) [inline Exiled.API/Features/Items/Armor.cs:93]:
```suggestion
                if (value is > 100 or < 0)
```

File: Exiled.API/Features/Items/Armor.cs [hunk 2]
```diff
@@ -84,32 +85,36 @@ public bool RemoveExcessOnDrop
...
         public int VestEfficacy
         {
             get => Base.VestEfficacy;
             set
             {
-                if (value is <= 101 and >= 0)
-                    Base.VestEfficacy = value;
-                else
-                    throw new ArgumentOutOfRangeException(nameof(VestEfficacy), "Value of armor efficacy must be between 0 and 100.");
+                if (value is > 101 or < 0)
```
Comment (Misaka-ZeroTwo) [inline Exiled.API/Features/Items/Armor.cs:111]:
```suggestion
                if (value is > 100 or < 0)
```

File: Exiled.API/Features/Pickups/BodyArmorPickup.cs [hunk 3]
```diff
@@ -76,32 +77,36 @@ internal BodyArmorPickup(ItemType type)
...
         public int VestEfficacy
         {
             get => vestEfficacy;
             set
             {
-                if (value is <= 101 and >= 0)
-                    vestEfficacy = value;
-                else
-                    throw new ArgumentOutOfRangeException(nameof(VestEfficacy), "Value of armor efficacy must be between 0 and 100.");
+                if (value is > 101 or < 0)
```
Comment (Misaka-ZeroTwo) [inline Exiled.API/Features/Pickups/BodyArmorPickup.cs:103]:
```suggestion
                if (value is > 100 or < 0)
```

File: Exiled.API/Features/Pickups/BodyArmorPickup.cs [hunk 4]
```diff
@@ -76,32 +77,36 @@ internal BodyArmorPickup(ItemType type)
...
         public int HelmetEfficacy
         {
             get => helmetEfficacy;
             set
             {
-                if (value is <= 101 and >= 0)
-                    helmetEfficacy = value;
-                else
-                    throw new ArgumentOutOfRangeException(nameof(HelmetEfficacy), "Value of armor efficacy must be between 0 and 100.");
+                if (value is > 101 or < 0)
```
Comment (Misaka-ZeroTwo) [inline Exiled.API/Features/Pickups/BodyArmorPickup.cs:85]:
```suggestion
                if (value is > 100 or < 0)
```

**Review by Misaka-ZeroTwo — COMMENTED** [review]
should be 100 instead of 101?

File: Exiled.API/Features/Pickups/Pickup.cs [hunk 5]
```diff
@@ -485,10 +481,51 @@ public void UnSpawn()
...
         public override string ToString() => $"{Type} ({Serial}) [{Weight}] *{Scale}* |{Position}| -{IsLocked}- ={InUse}=";
+
+        /// <summary>
+        /// Returns the Pickup with the according property from the Item.
+        /// </summary>
+        /// <param name="item"> Item-related data to give to the Pickup.</param>
+        /// <returns>A Pickup containing the Item-related data.</returns>
+        internal virtual Pickup GetItemInfo(Items.Item item)
+        {
+            if (item is not null)
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Pickups/Pickup.cs:508]: Use conditional expression

File: Exiled.API/Features/Pickups/Pickup.cs [hunk 6]
```diff
@@ -485,10 +481,51 @@ public void UnSpawn()
...
+        }
+
+        /// <summary>
+        /// Returns the Item with the according property from the Pickup.
+        /// </summary>
+        /// <param name="item"> Pickup-related data to give to the Item.</param>
+        /// <returns>A Item containing the Pickup-related data.</returns>
+        internal virtual Items.Item GetPickupInfo(Items.Item item)
+        {
+            if (item is not null)
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Pickups/Pickup.cs:523]: Same as above

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

**Pull request merged**